### PR TITLE
Nektarinchik/db scripts

### PR DESCRIPTION
**Pull request by nikita-ermolaev-coral**
Added:

Tables: UserTask, TaskTrack, TaskState

Views: vUserTask

SeedData: TaskStateSeedData

Procedures: SetUserTaskAsSuccess, SetUserTaskAsFail, DeleteTask (@temaors)

### CODE REVIEW

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿CREATE TABLE [dbo].[UserTask]
+(
+	[UserTaskId] INT NOT NULL IDENTITY(1, 1),
+	[TaskId]     INT NOT NULL,
+	[UserId]     INT NOT NULL,
+	[StateId]    INT NOT NULL,
+
+	CONSTRAINT PK_UserTask_UserTaskId PRIMARY KEY (UserTaskId),
+	CONSTRAINT FK_UserTask_TaskState_StateId FOREIGN KEY (StateId) REFERENCES TaskState (StateId)
+		ON DELETE CASCADE
```
Comment (dbogdan2000) [inline DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql:10]: Add Update on cascade, please

Comment (dbogdan2000) [inline DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql:10]: Add foreign keys with references on UserProfiles table and Tasks table, please

File: DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTrack.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿CREATE TABLE [dbo].[TaskTrack]
+(
+	[TaskTrackId] INT  NOT NULL IDENTITY (1, 1),
+	[UserTaskId]  INT  NOT NULL,
+	[TrackDate]   DATE NOT NULL,
+	[TrackNote]   NVARCHAR(255) NOT NULL,
+
+	CONSTRAINT PK_TaskTrack_TaskTrackId PRIMARY KEY (TaskTrackId),
+	CONSTRAINT FK_TaskTrack_UseTask_UserTaskId FOREIGN KEY (UserTaskId) REFERENCES UserTask (UserTaskId)
+		ON DELETE CASCADE
```
Comment (dbogdan2000) [inline DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTrack.sql:10]: Add Update on cascade, please

File: DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsSuccess.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿CREATE PROCEDURE [dbo].[SetUserTaskAsSuccess]
+	@UserId int,
+	@TaskId int
+AS
+BEGIN
```
Comment (dbogdan2000) [inline DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsSuccess.sql:5]: Why have you added begin and end in the procedure?

File: DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsFail.sql [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿CREATE PROCEDURE [dbo].[SetUserTaskAsFail]
+	@UserId int,
+	@TaskId int
+AS
+BEGIN
```
Comment (dbogdan2000) [inline DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsFail.sql:5]: Why have you added begin and end in the procedure?

File: DIMS-Core.Database/DIMS-Core.Database/Procedures/DeleteUser.sql [hunk 5]
```diff
@@ -0,0 +1,7 @@
+﻿CREATE PROCEDURE [dbo].[DeleteUser]
+	@UserId INT
+AS
+BEGIN
```
Comment (dbogdan2000) [inline DIMS-Core.Database/DIMS-Core.Database/Procedures/DeleteUser.sql:4]: Why have you added begin and end in the procedure?

File: DIMS-Core.Database/DIMS-Core.Database/Procedures/DeleteTask.sql [hunk 6]
```diff
@@ -0,0 +1,6 @@
+﻿CREATE PROCEDURE [dbo].[DeleteTask]
+    @TaskId INT
+AS
+BEGIN
```
Comment (dbogdan2000) [inline DIMS-Core.Database/DIMS-Core.Database/Procedures/DeleteTask.sql:4]: Why have you added begin and end in the procedure?

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql (hunk 1 above)
Comment (dbogdan2000) [inline DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql:10]: Add Update on cascade, please

File: DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsSuccess.sql (hunk 3 above)
Comment (nikita-ermolaev-coral) [reply DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsSuccess.sql:5]: I have added BEGIN and END in the procedure to separate procedure's body from another code.
Like in tutorial:
[https://metanit.com/sql/sqlserver/11.1.php](url)

File: DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsFail.sql (hunk 4 above)
Comment (nikita-ermolaev-coral) [reply DIMS-Core.Database/DIMS-Core.Database/Procedures/SetUserTaskAsFail.sql:5]: I have added BEGIN and END in the procedure to separate procedure's body from another code.
Like in tutorial:
[https://metanit.com/sql/sqlserver/11.1.php](url)

File: DIMS-Core.Database/DIMS-Core.Database/Procedures/DeleteUser.sql (hunk 5 above)
Comment (nikita-ermolaev-coral) [reply DIMS-Core.Database/DIMS-Core.Database/Procedures/DeleteUser.sql:4]: I have added BEGIN and END in the procedure to separate procedure's body from another code.
Like in tutorial:
[https://metanit.com/sql/sqlserver/11.1.php](url)

File: DIMS-Core.Database/DIMS-Core.Database/Procedures/DeleteTask.sql (hunk 6 above)
Comment (nikita-ermolaev-coral) [reply DIMS-Core.Database/DIMS-Core.Database/Procedures/DeleteTask.sql:4]: I have added BEGIN and END in the procedure to separate procedure's body from another code.
Like in tutorial:
[https://metanit.com/sql/sqlserver/11.1.php](url)

**Review by dbogdan2000 — APPROVED** [review]

**Review by temaors — APPROVED** [review]
Approved

**Pull request merged**